### ISSUE TITLE
Fix normalization leakage in exp12_residual_vq.py data loading

### ISSUE THREAD

**Issue by yelabb** [yelabb/PhantomX#1]
## Problem

The data loading section in `exp12_residual_vq.py` (lines 729-730) computes normalization statistics on the **entire dataset** before the train/val split:

```python
# Current (LEAKY):
spike_counts = (spike_counts - spike_counts.mean(0)) / (spike_counts.std(0) + 1e-6)
velocities = (velocities - velocities.mean(0)) / (velocities.std(0) + 1e-6)

window_size = 10
dataset = SlidingWindowDataset(spike_counts, velocities, window_size)

n_train = int(0.8 * len(dataset))
train_dataset = torch.utils.data.Subset(dataset, list(range(n_train)))
val_dataset = torch.utils.data.Subset(dataset, list(range(n_train, len(dataset))))
```

This causes **data leakage** because:
1. Validation data statistics contaminate training normalization
2. Model has indirect access to validation distribution during training
3. Reported R² scores are optimistically biased

This was already identified in Exp 18/19 (LADR-VQ) where correcting this issue caused a significant performance drop (R² 0.784 → 0.695), suggesting the original R² = 0.776 is inflated.

## Proposed Fix

Rewrite the data loading to compute normalization **only on training data**:

```python
# Proposed (CORRECT):
window_size = 10
n_samples = len(spike_counts) - window_size + 1
n_train = int(0.8 * n_samples)

# Split FIRST, normalize SECOND
train_indices = list(range(n_train))
val_indices = list(range(n_train, n_samples))

# Compute stats on training data only
train_spikes_raw = spike_counts[:n_train + window_size - 1]
train_vel_raw = velocities[window_size - 1:n_train + window_size - 1]

spike_mean = train_spikes_raw.mean(0)
spike_std = train_spikes_raw.std(0) + 1e-6
vel_mean = train_vel_raw.mean(0)
vel_std = train_vel_raw.std(0) + 1e-6

# Apply to full dataset using TRAINING stats
spike_counts_norm = (spike_counts - spike_mean) / spike_std
velocities_norm = (velocities - vel_mean) / vel_std

dataset = SlidingWindowDataset(spike_counts_norm, velocities_norm, window_size)
train_dataset = torch.utils.data.Subset(dataset, train_indices)
val_dataset = torch.utils.data.Subset(dataset, val_indices)
```

## Additional Considerations

1. **Save normalization stats** with the model checkpoint for inference
2. **Update other experiment files** (exp9, exp10, exp11) with the same fix
3. **Re-run exp12 benchmark** to get unbiased R² scores
4. **Document the expected performance drop** in RESEARCH_LOG.md

## Impact

This fix may reduce reported R² from 0.776 to ~0.70-0.72 (based on Exp 19 findings), but will provide **honest, reproducible benchmarks** that are valid for real-world deployment.

## Files to Update

- [ ] `python/exp12_residual_vq.py` (primary)
- [ ] `python/exp9_progressive_vqvae.py`
- [ ] `python/exp10_beat_lstm.py`
- [ ] `python/exp11_close_gap.py`
- [ ] `RESEARCH_LOG.md` (update results after re-benchmarking)
